src/Chain.php

```diff
@@ -311,6 +313,27 @@ public function allowEmpty($allowEmpty)
     }
 
     /**
+     * Attach a representation of this Chain to the Output\Structure $structure.
+     *
+     * @internal
+     * @param Structure $structure
+     * @param MessageStack $messageStack
+     * @return Structure
+     */
+    public function output(Structure $structure, MessageStack $messageStack)
+    {
+        $subject = new Subject($this->key, $this->name);
+
+        foreach ($this->rules as $rule) {
+            $rule->output($subject, $messageStack, $this->key, $this->name);
```

Comment:
You put the key and the name in here as separate parameters, yet you already have those parameters in `$subject`, is there a reason to have those separate?